app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,20 +72,142 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfFirstNameEditTextIsInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
+    public void signIn_checkIfLastInitialEditTextIsInitiallyEmpty() {
+        onView(withId(last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+    }
+
+    @Test
+    public void signIn_chooseAvatar_01() {
```

Comment:
[`onData`](https://developer.android.com/reference/android/support/test/espresso/Espresso.html#onData(org.hamcrest.Matcher<? extends java.lang.Object>)) works on data not on views, hence the name.
There's [more documentation available](https://google.github.io/android-testing-support-library/docs/espresso/basics/index.html#get-started-with-a-simple-test-using-ondata) on how to use it.